src/scenes/Projects/Budget/ProjectBudget.tsx

```diff
@@ -0,0 +1,176 @@
+      amount: String(dollarAmount),
+      canEdit,
+    };
+    return rows.concat(row);
+  }, []);
+
+  const columns = [
+    {
+      title: 'ID',
+      field: 'id',
+      hidden: true,
+    },
+    {
+      title: 'Organization',
+      field: 'organization',
+      editable: 'never' as const,
+    },
+    {
+      title: 'Fiscal Year',
+      field: 'fiscalYear',
+      editable: 'never' as const,
+      render: (rowData: BudgetRowData) => `FY${rowData.fiscalYear}`,
+    },
+    {
+      title: 'Amount',
+      field: 'amount',
+      type: 'currency' as const,
+      editable: (_: unknown, rowData: BudgetRowData) => rowData.canEdit,
+      cellStyle: {
+        display: 'flex',
+        flexDirection: 'row' as const,
+        justifyContent: 'flex-end',
+      },
+    },
+    {
+      title: 'Can Edit',
+      field: 'canEdit',
+      hidden: true,
+    },
+  ];
+
+  return (
+    <Content>
+      {error ? (
+        <Typography variant="h4">Error fetching Project Budget</Typography>
+      ) : canReadBudget === false ? (
+        <Typography variant="h4">
+          You do not have permission to view this project's budget
+        </Typography>
+      ) : (
+        <>
+          {loading ? (
+            <Skeleton variant="text" width="20%" />
+          ) : (
+            <Breadcrumbs>
+              <Breadcrumb to="/projects">Projects</Breadcrumb>
+              <Breadcrumb to={`/projects/${projectId}`}>
+                {projectName}
+              </Breadcrumb>
+            </Breadcrumbs>
+          )}
+          <header className={classes.header}>
+            {loading ? (
+              <>
+                <Skeleton variant="text" width="20%" />
+                <Skeleton variant="text" width="10%" />
+              </>
+            ) : (
+              <>
+                <Typography variant="h2">
+                  {data?.project.name.value} Budget
+                </Typography>
+                {budget && (
+                  <Typography variant="body1" className={classes.total}>
+                    Total: {formatCurrency(budget.total)}
+                  </Typography>
+                )}
+              </>
+            )}
+          </header>
+          <section className={classes.tableWrapper}>
+            <Table
+              data={rowData}
+              columns={columns}
+              isLoading={loading}
+              cellEditable={
+                canEditBudget
+                  ? {
+                      onCellEditApproved: async (newAmount, _, data) => {
+                        const input = {
+                          budgetRecord: {
+                            id: data.id,
+                            amount: Number(newAmount),
+                          },
+                        };
+                        await updateBudgetRecord({
+                          variables: { input },
+                          refetchQueries: [GQLOperations.Query.ProjectBudget],
+                        });
```

Comment:
I believe this won't update the total until the query is refetched "some time later". Could we either:
- calculate the total ourselves based on the records which should update immediately in cache
- or write to the apollo cache directly the new total. I guess that still requires us to calculate it first.

Reply:
Refetches for live queries run immediately upon successful completion of the mutation; this is the whole reason for their existence. The only reason it would take a noticeably long time to update the data would be because of network, server, or db latency. If we're worried about that, then the next option would generally be updating the cache using the returned data. This would **not** require us to calculate it, provided we request the new amount in the returned mutation data. But it is sort of a pain, which is why we have been sticking with refetches so far.

Reply:
I'm worried about API performance yes.
If we didn't want to calculate it we could add a `total` property to the `UpdateBudgetRecordOutput` output type. However, I don't think Apollo Cache would connect that to `Budget.total` automatically so we'd have to write it manually.

I think calculating it ourselves it honestly probably the easiest. The `total` from the API was really there as a helper for list or summary views that weren't getting all the records.
It's only a one-liner...
https://github.com/SeedCompany/cord-api-v3/blob/a32c02431f71febad76d8f89debea17f770488a8/src/components/budget/budget.resolver.ts#L54

Thoughts?

Reply:
Yeah, fair enough. It does seem simpler to just calculate the total rather than writing a second value back into the cache alongside the updated amount